modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/README.md

```diff
@@ -0,0 +1,119 @@
+- [Search Pipelines](#search-pipelines)
+  - [Architecture](#architecture)
+  - [Search Processors](#search-processors)
+    - [Creating a Search Processor](#creating-a-search-processor)
+    - [Creating a Pipeline](#creating-a-search-pipeline)
+
+# Search Pipelines
+
+This README briefly covers the two types of search processors, explains how you can use them to create search pipelines, and walks through the creation of a new processor.
+
+## Architecture
+
+Search pipelines allow cluster operators to create and reuse [components](#search-processors) to transform search queries and results.
+
+With search pipelines, the operator can combine multiple [search processors](#search-processors) to create a transform which acts on the search request and/or search response.
+
+The primary benefit of search pipelines is that they live entirely inside OpenSearch and therefore require less development time. Further, search pipelines support standard APIs for accessing metrics and disaster recovery.
+
+## Search Processors
+
+You can create many search pipelines by combining search processors in various orders. There are two types of search processors:
+
+1. search request processors which transform a request _before_ it is executed;
+2. search response processors which transform the output of a request _after_ it is executed.
+
+Currently, there are two search request and two search response processors.
```

Comment:
Do we want to enumerate all of the search-pipeline-common search processors here? 

We'd need to make sure that list doesn't go out of date (e.g. I added three new processors in https://github.com/opensearch-project/OpenSearch/pull/9405).

Alternatively, for this README (which I assume should be more developer focused), we could point them to the `SearchPipelineCommonModulePlugin.java` class and the registered processor classes.

Reply:
For these I just took the description from the documentation website. I figured it may be helpful to include in them in the dev guide about the feature as well. 

If we don't want to worry about them going out of date, we could point them to the class you mentioned and also the doc website where they should all be listed--I will go this route for now.